linera-core/src/client/mod.rs

```diff
@@ -2550,7 +2550,13 @@ where
             ));
         };
         if manager.can_propose(&identity, round) {
-            let certificate = self.propose_block(block.clone(), round, manager).await?;
+            let result = self.propose_block(block.clone(), round, manager).await;
+            if let Err(ChainClientError::LocalNodeError(_)) = result {
+                // If the local node rejected the proposal, clear the pending block
+                // since we don't want to retry in this case.
+                self.clear_pending_block();
```

Comment:
The `block` above might come from `fn hightest_validated_block` which is either `requested_locked` or `requested_proposed`. But in both cases the block should have been validated already?

Reply:
True, but if it's a validated block, why would the local node fail when trying to propose it? Technically it has already went through the local node and succeeded. Should not be a reachable case AFAIU 🤔
But I can add a check here so that we only clear it if we were actually just trying to propose it and failed